server/src/main/java/keywhiz/utility/DSLContexts.java

```diff
@@ -40,12 +40,9 @@ public static DSLContext databaseAgnostic(DataSource dataSource) throws SQLExcep
     try (Connection conn = dataSource.getConnection()) {
       dialect = dialect(conn);
     }
-    if (dialect == SQLDialect.H2) {
-      return DSL.using(dataSource, SQLDialect.H2,
+    return DSL.using(dataSource, SQLDialect.H2,
```

Comment:
you don't want to use SQLDialect.H2 for non H2 connections. We want DSL.using(dataSource, new Settings()...) instead.